lib/routemaster/api_client.rb

```diff
@@ -25,16 +25,23 @@
 
 module Routemaster
   class APIClient
+
+    # Memoize the root resources at Class level so that we don't hit the cache
+    # all the time to fetch the root resource before doing anything else.
+    @@root_resources = {}
```

Comment:
Yes, we need to have an actual discussion about how to `resource`-ify the root endpoints (or find another way to bust the cache when the resource changes.